llarp/dht/messages/findintro.cpp

```diff
@@ -105,20 +107,23 @@ namespace llarp
           // find closer peer
           if(!dht.Nodes()->FindClosest(target, closer))
             return false;
-          if(relayed)
-            dht.LookupIntroSetForPath(S, T, pathID, closer);
-          else
-            replies.emplace_back(new GotIntroMessage(From, closer, T));
+          replies.emplace_back(new GotIntroMessage(From, closer, T));
           return true;
         }
 
         Key_t us     = dht.OurKey();
         Key_t target = S.ToKey();
         // we are recursive
-        if(dht.Nodes()->FindCloseExcluding(target, peer, exclude))
+        const auto rc = dht.GetRouter()->nodedb()->FindClosestTo(target);
         {
+          peer = Key_t(rc.pubkey);
+          if(peer == us)
+          {
+            replies.emplace_back(new GotIntroMessage({}, T));
+            return true;
+          }
```

Comment:
If we are called to be recursive then we need to consider that the closest router may not have the record, even if the closest router is us.  (Maybe we were temporarily done when the introset got published, or maybe we are a brand new router and weren't active 2 minutes ago when the introset got published).